x-pack/plugins/ml/public/ui_actions/open_in_anomaly_explorer_action.tsx

```diff
@@ -122,13 +199,12 @@ export function createOpenInExplorerAction(
         await application.navigateToUrl(anomalyExplorerUrl!);
       }
     },
-    async isCompatible({
-      embeddable,
-    }: SwimLaneDrilldownContext | AnomalyChartsFieldSelectionContext) {
+    async isCompatible({ embeddable }: EmbeddableApiContext) {
+      if (!isApiCompatible(embeddable)) return false;
```

Comment:
nit: Here, wouldn't it be simpler to call your more specific type guards like:

`return isSwimLaneEmbeddableContext(context) || isAnomalyChartsEmbeddableContext(context)`? That way you wouldn't need to have the separate `isApiCompatible` method which checks the same things?

Reply:
I first wrote this `isCompatible` method looking at the example and added custom type guards later. Updated in https://github.com/elastic/kibana/pull/178729/commits/dfaef990f4e791fa1219c5f39640ee83bf99d6bb